### DISCUSSION TITLE
详解 Kubernetes Volume 的实现原理 · /kubernetes-volume

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#538] [Announcements]
https://draveness.me/kubernetes-volume

**Comment by JumpJumpSparrow** [comment]
想问一下，这样的个人博客是怎么搭建起来的

**Comment by draveness** [comment]
> @JumpJumpSparrow
> 想问一下，这样的个人博客是怎么搭建起来的

jekyll

**Comment by JumpJumpSparrow** [comment]
谢谢> @draveness
> > @JumpJumpSparrow
> > 想问一下，这样的个人博客是怎么搭建起来的
> 
> jekyll

**Comment by TiantianUpup** [comment]
想问一下博主工作方向是虚拟化这块的吗

**Comment by draveness** [comment]
> @TiantianUpup
> 想问一下博主工作方向是虚拟化这块的吗

并不是，目前主要是做基础架构和服务治理

**Comment by TheHati** [comment]
您好，想请问一下configMap或者secret可以以挂载卷的形式挂载给容器中的普通用户吗？因为一旦这两类对象只要挂载进去，对应文件夹就会变成只读文件系统。而生产环境中又需要在非root用户下开展，谢谢了，能不能帮忙解答一下。

**Comment by draveness** [comment]
> @TheHati
> 您好，想请问一下configMap或者secret可以以挂载卷的形式挂载给容器中的普通用户吗？因为一旦这两类对象只要挂载进去，对应文件夹就会变成只读文件系统。而生产环境中又需要在非root用户下开展，谢谢了，能不能帮忙解答一下。

挂载给普通用户是什么意思？

**Comment by TheHati** [comment]
> @draveness
> > @TheHati
> > 您好，想请问一下configMap或者secret可以以挂载卷的形式挂载给容器中的普通用户吗？因为一旦这两类对象只要挂载进去，对应文件夹就会变成只读文件系统。而生产环境中又需要在非root用户下开展，谢谢了，能不能帮忙解答一下。
> 
> 挂载给普通用户是什么意思？

就是挂载文件不是root root属性，而是username groupname属性。我并不是想修改readonly system性质(也改不了)，只是在挂载之前想设置所有权和所属组。我接触k8s不久，不知道有没有表述清楚，麻烦啦

**Comment by bzd111** [comment]
最后相关文章那里展示有问题

**Comment by Formergg** [comment]
### typo

```diff
-对应的文件目录就会通过 `MkdirAl` 方法创建
+对应的文件目录就会通过 `MkdirAll` 方法创建
```


---

*2020-07-30 UPDATES: 已修复*

**Comment by FingerLiu** [comment]
图片都渲染不出来了。

**Comment by chenggui53** [comment]
> @FingerLiu
> 图片都渲染不出来了。

+1 markdown mermaid 的图全部没有渲染了

**Comment by happy-v587** [comment]
HostPath 实现原理是怎么样的？文中好像没有提及到

**Comment by pzghost** [comment]
- 哥，该更新啦。
访问模式这里可以加上 [ReadWriteOncePod](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes)
- 哥，这里是不是有些出入：
回收策略还差一个 [Recycle](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#reclaim-policy)(虽然已经deprecated了), 这里的图片好像也贴错了吧："dynamic provisioning"
- 哥，建个微信群吧，给大家指导指导生涯。
